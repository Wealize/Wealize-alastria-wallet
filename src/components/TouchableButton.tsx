import React, { useState } from 'react'
import {
  TouchableHighlight,
  View,
  Text,
  GestureResponderEvent
} from 'react-native'

import { Colors } from '../utils/themes'
import StylesButton from './styles/TouchableButton.styles'

const TouchableButton = ({
  title,
  disabled,
  onPress
}: {
  title?: string
  disabled?: boolean
  onPress: (event: GestureResponderEvent) => void
}) => {
  const [isPressed, setIsPressed] = useState(false)

  const selectStyle = () => {
    if (disabled) {
      return StylesButton.btnDisabled
    }
    return isPressed ? StylesButton.btnPress : StylesButton.btnNormal
  }

  const touchProps = {
    activeOpacity: 1,
    underlayColor: Colors.pressedMainButton,
    style: selectStyle(),
    onHideUnderlay: () => setIsPressed(false),
    onShowUnderlay: () => setIsPressed(true)
  }

  return (
    <View style={StylesButton.container}>
      <TouchableHighlight disabled={disabled} onPress={onPress} {...touchProps}>
        <Text style={StylesButton.text}>{title}</Text>
      </TouchableHighlight>
    </View>
  )
}

export default TouchableButton
