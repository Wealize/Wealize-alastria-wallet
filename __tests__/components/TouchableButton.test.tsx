import React from 'react'
import { render } from '@testing-library/react-native'

import TouchableButton from '../../src/components/TouchableButton'

describe('TouchableButton', () => {
  it('should render correctly TouchableButton', () => {
    const result = render(
      <TouchableButton
        title="title"
        disabled={false}
        onPress={() => 'pressed'}
      />
    )

    expect(result).toMatchSnapshot()
  })
})
