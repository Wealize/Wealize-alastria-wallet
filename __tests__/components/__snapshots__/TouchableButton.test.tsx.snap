// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TouchableButton should render correctly TouchableButton 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "flex-end",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ca2b63",
        "borderRadius": 45,
        "height": 39,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 224,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 16,
        }
      }
    >
      title
    </Text>
  </View>
</View>
`;
